handler/src/test/java/io/netty/handler/timeout/IdleStateHandlerTest.java

```diff
@@ -391,5 +417,32 @@ public Object consume() {
             }
             return null;
         }
+
+        /**
+         * Consume the part of a message.
+         *
+         * @param byteCount count of byte to be consumed
+         * @return the message If all content is consumed, otherwise return {@code null}
+         */
+        public Object consumePart(int byteCount) {
+            ChannelOutboundBuffer buf = unsafe().outboundBuffer();
+            if (buf != null) {
+                Object msg = buf.current();
+                if (msg instanceof ByteBuf) {
+                    ByteBuf byteBuf = (ByteBuf) msg;
+                    if (byteBuf.readableBytes() > byteCount) {
+                        buf.removeBytes(byteCount);
+                        return null;
+                    } else {
+                        ReferenceCountUtil.retain(msg);
+                        buf.removeBytes(byteCount);
+                        return msg;
+                    }
+                } else {
```

Comment:
nit you can remove the else as you return in the previous if block

Reply:
Done